components/bluetooth-sink-switch/README.md

```diff
@@ -0,0 +1,276 @@
+        device "C4:FB:20:63:A7:F2"
+        profile "a2dp"
+        delay -20000
+    }
+    hint {
+        show on
+        description "Gesas Headphones"
+    }
+}
+~~~
+
+In case of doubt, reboot.
+
+Test the new audio device (mine is called `pcm.btheadphone`). Also test the soundcard (here `pcm.hifiberry` for the regular speakers).
+~~~sh
+$ aplay -D btheadphone /usr/share/sounds/alsa/Front_Center.wav 
+$ aplay -D hifiberry /usr/share/sounds/alsa/Front_Center.wav
+~~~
+
+#### Step 2) Setting up mpd.conf
+
+You need to set up two audio_output sections. **The order is important**: the first entry must relate to the soundcard setup, the second entry must relate to the bluetooth setup. Give meaningful names, as they will show up in the Web Interface.
+
+~~~
+# The first entry should match your soundcard. If you have a working setup, there is most likly no need to change it!
+# As an exanple, here is my configuration for my HifiBerry MiniAmp.
+audio_output {
+        type             "alsa"
+        name             "HifiBerry Speakers"   # This name will show up in the Web Interface Status 
+        device           "hifiberry"            # This is the pcm.hifiberry device from the asound.conf. 
+                                                # If you did not specify a name, but use the 'default', delete this entry
+        auto_resample    "no"                   # Depends on your asound.conf. In doubt delete this line
+        mixer_control    "Master"               # This is the iFace name, you will recognize from your Phoniebox installation
+}
+
+# The second entry belongs to the bluetooth device
+audio_output {
+        type            "alsa"
+        name            "Gesas Headphones" # This name will show up in the Web Interface Status 
+        device          "btheadphone"      # This is the pcm.btheadphone device from the asound.conf
+        mixer_type      "software" 
+        auto_resample   "no"
+        auto_format     "no"
+        enabled         "no"               # Default is disabled, keepin the soundcard as primary output
+}
+~~~
+
+Restarting the mpd.service or else reboot. 
+
+Check the setup:
+
+~~~sh
+$ mpc outputs
+Output 1 (HifiBerry Speakers) is disabled
+Output 2 (Gesas Headphones) is enabled
+~~~
+
+You may switch with `$ mpc enable only 1` and `$ mpc enable only 2`. Play some music and use these commands to check you mpd configuration. You should be able to switch the audio output between the two devices.
+
+#### Step 3) Run the installer
+
+This sets up the appropriate user rights and registers the component with global settings etc.
+
+~~~sh
+$ cd components/bluetooth-sink-switch
+$ ./install-bt-sink-switch.sh
+~~~
+
+
+#### Step 4) Fine-tuning
+
+**Status LED**
+
+An optional status LED will be turned on if the audio sink is set to bluetooth. If a toggle command is issued, but no bluetooth device is connected, the LED will blink three times. Looks very neat, if you have a button with integrated LED. 
+
```

Comment:
I think it would make sense to use GPIO control (which already has led function) or gpio_settings.ini for this setting. We introduced gpio_setttings.ini to have one central place for GPIO configuration

Reply:
Do you mean, 
-  Use the gpio_settings.ini to read the GPIO number for the LED from? Is there an API that already does this for me, which I could use?
- Use the gpio API to switch an LED on/off? 

In any case, when using gpio_settings.ini, an LED device is opened in the process gpio_control.py. But I want to control the LED form bt-sink-switch.py (to avoid having to poll bt-sink-switch status from gpio_control.py). This will potentially lead to a resource conflict of the LED pin. How do I resolve that?

Reply:
@veloxidSchweiz, do you have an idea?